File: python/p0100.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        # BASE CASE
        if not p and not q:
            return True

        if not p or not q:
            return False

        if p.val != q.val:
            return False

        # RECURSIVE CASE
        return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)
        
        

File: javascript/p0100.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function (p, q) {
  // BASE CASE
  if (!p && !q) return true;
  if (!p || !q) return false;
  if (p.val !== q.val) return false;

  // RECURSIVE CASE
  return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
};
